PSSwagger/Get-ArmResourceIdParameterValue.ps1

```diff
@@ -0,0 +1,46 @@
+Microsoft.PowerShell.Core\Set-StrictMode -Version Latest
+
+<#
+.DESCRIPTION
+   Get parameter values from the specified Azure resource identifier.
+
+.PARAMETER  Id
+    The unique resource identifier of the Azure Resource.
+
+.PARAMETER  IdTemplate
+    The unique resource identifier template of the Azure Resource.
+#>
+function Get-ArmResourceIdParameterValue {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Id,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $IdTemplate
+    )
+
+    $IdTokens = $Id.Split('/', [System.StringSplitOptions]::RemoveEmptyEntries)
+    $IdTemplateTokens = $IdTemplate.Split('/', [System.StringSplitOptions]::RemoveEmptyEntries)
+    
+    if ($IdTokens.Count -ne $IdTemplateTokens.Count) {
+        Write-Error -Message ("Invalid resource id '{0}'. Expected id template is '{1}'." -f $Id, $IdTemplate) -ErrorId 'InvalidResourceIdValue'
+        return
+    }
+    
+    $ResourceIdParameterValues = [ordered]@{}
+    for ($i = 0; $i -lt $IdTemplateTokens.Count; $i++) {
+        if ($IdTokens[$i] -ne $IdTemplateTokens[$i]) {
+            if ($IdTemplateTokens[$i] -notmatch '{*}') {
```

Comment:
>{*} [](start = 49, length = 3)

What's the goal of this pattern in plain English? Assuming it's what I think, should be: '^\{.*\}$'.

'{blah}' -replace '{*}','x' should result in '{blahx'